### ISSUE TITLE
Unable to build a brand new .NET MAUI Balzor hybrid App using .NET 8   

### ISSUE THREAD

**Issue by MohammedBen** [dotnet/maui#24680]
### Description

I have been unable to successfully build a brand new .NET MAUI Blazor hybrid App using .NET 8 without getting a 
"The process cannot access the file "located in obj\Debug\net8.0-android/resizetizer/............................/appicon.png" because it is being used by another process. It is generated from the following file, Microsoft.Maui.Resizetizer.After.Targets error.
Same issue is happening when I target iOS and try to build against iOS. 

I have tried all the suggested twisted brilliant ideas like delete all obj and debug folders. Restart everything you can imagine! remove read-only flags .... etc.

am I missing something? am I the only one?  can't be ! 
is it normal that I cannot build a brand new project that is out of the box. I am literally using a fresh template, not a single piece of code added !   

### Steps to Reproduce

simple create, build new project

### Link to public reproduction project repository

none

### Version with bug

8.0.82 SR8.2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.49

### Affected platforms

iOS, Android

### Did you find any workaround?

no, it only works in .NET 7

### Relevant log output

```shell
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	MAUI0000	There was an exception processing the image 'C:\Users\XXXXXXXXXXX\source\repos\MauiTestApp\MauiTestApp\Resources\AppIcon\appicon.svg': System.IO.IOException: The process cannot access the file 'C:\Users\XXXXXXXXXXXXXXXXX\source\repos\MauiTestApp\MauiTestApp\obj\Debug\net8.0-android\resizetizer\r\mipmap-xxhdpi\appicon.png' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at Microsoft.Maui.Resizetizer.SkiaSharpAppIconTools.Save(SKBitmap tempBitmap, String destination, Func`1 getStream) in /_/src/SingleProject/Resizetizer/src/SkiaSharpAppIconTools.cs:line 75
   at Microsoft.Maui.Resizetizer.SkiaSharpAppIconTools.Resize(DpiPath dpi, String destination, Func`1 getStream) in /_/src/SingleProject/Resizetizer/src/SkiaSharpAppIconTools.cs:line 54
   at Microsoft.Maui.Resizetizer.ResizetizeImages.ProcessAppIcon(ResizeImageInfo img, ConcurrentBag`1 resizedImages) in /_/src/SingleProject/Resizetizer/src/ResizetizeImages.cs:line 229
   at Microsoft.Maui.Resizetizer.ResizetizeImages.<>c__DisplayClass34_0.<ExecuteAsync>b__0(ResizeImageInfo img) in /_/src/SingleProject/Resizetizer/src/ResizetizeImages.cs:line 65	MauiTestApp (net8.0-android)	C:\Users\XXXXXXXXXXXXX\.nuget\packages\microsoft.maui.resizetizer\8.0.72\buildTransitive\Microsoft.Maui.Resizetizer.After.targets	631
```

**Comment by RoiChen001** [comment]
This problem cannot be reproduced on the latest 17.12.0 Preview 1.0(8.0.82 & 8.0.80 & 7.0.101)

**Comment by MohammedBen** [comment]
It still fails with the latest version as shown below:

![image](https://github.com/user-attachments/assets/8892b53e-9bd9-4ef3-b476-d0d62204e8a4)

![image](https://github.com/user-attachments/assets/3cc65ed0-ed53-43a7-8ae5-7c0776b84a89)

**Comment by MohammedBen** [comment]
To add to the mix, I create another new Maui hybrid Blazor project using VS 17.12 Preview 1. 
Target android emulator -> build succeeds and loads in emulator successfully. 
Target iOS simulator -> builds succeeds and unable to load in sumilator.

So, I close VS, re-open the project again, try to build and sure enough I get the same error shown the image above !
This is unbelievable. I can't even get an empty built-in template to build and deploy !

How am I supposed to migrate my app from .NET 7 MAUI to .NET 8 MAUI !

**Comment by mattleibow** [comment]
Do you perhaps have an anti-virus that could be locking the file? If you close VS and build from the command line with `dotnet build` does anything change?

**Comment by mattleibow** [comment]
@jonathanpeppers thoughts?

**Comment by MohammedBen** [comment]
@mattleibow, I'm getting the same errors with a normal non-blazor Maui App. 
I'm looking at the anti virus option, but so far I don't see anything obvious

**Comment by jonathanpeppers** [comment]
@MohammedBen you will need to use a tool like [`Process Explorer`](https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer) to find out what process is locking the file.

I recently added something to help in the future:
* https://github.com/dotnet/android-tools/pull/242

But this won't ship until .NET 9 RC 2.

@mattleibow you might consider making using of the new MSBuild API `LockCheck.GetLockedFileMessage()` in MAUI takss. My change above will only help Android MSBuild tasks.

**Comment by MohammedBen** [comment]
I have a few updates as follows,

I was able to disable our organization's anti-virus (TEMPORARILY). I needed to close all the VS instances and delete obj and debug folders, and sure enough I am able to build successfully and deploy to both environments (android and iOS).

This is still an issue for us as I am having to stop an antivirus program that is supposed to be enabled per our organization's policy. I am not sure what is the .NET MAUI build process doing behind the scenes as I have been working with all kinds of projects from blazor projects to ASP.net core apps to desktop apps and never had any issues. 
Matter of fact, we have upgraded from XAMARIN to .net MAUI Hybrid and never came across such issue with XAMARIN.
